src/Http/Http.Abstractions/src/IBindableFromHttpContextOfT.cs

```diff
@@ -0,0 +1,22 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Reflection;
+
+namespace Microsoft.AspNetCore.Http;
+
+/// <summary>
+/// Defines a mechanism for creating an instance of a type from an <see cref="HttpContext"/> when binding parameters for an endpoint
+/// route handler delegate.
+/// </summary>
+/// <typeparam name="TSelf">The type that implements this interface.</typeparam>
+public interface IBindableFromHttpContext<TSelf> where TSelf : class?, IBindableFromHttpContext<TSelf>
```

Comment:
Did `notnull` like @BrennanConroy suggested not work?

Reply:
you can't combine `notnull` and `class` so if we want to limit to classes like you said then we need to keep it as `class`